Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1201,6 +1218,10 @@ static AffineTransform documentSpaceToPageSpaceTransform(const IntDegrees& pageR
             return;
         if (selectedItemTag)
             performContextMenuAction(toContextMenuItemTag(selectedItemTag.value()));
+        m_selectionTrackingData.lastHandledEventWasContextMenuEvent = true;
+        // We make a repaint request if we've finished a context menu interaction.
+        // This allows us to switch to an inactive appearance for the selections.
+        m_contentsLayer->setNeedsDisplay();
```

Comment:
Your comment could turn into self-documenting code if you wrapped up these two lines in a function, and had a corresponding function for the place where you had to comment:
```
        // Selections should be presented as inactive if we just interacted with a context menu.
        bool isVisibleAndActive = !m_selectionTrackingData.lastHandledEventWasContextMenuEvent;
```

So you could have:
`repaintOnSelectionActiveStateChange()` and `selectionActiveForContextMenu()` or something.

Reply:
Thanks for the suggestion. I'll try it. I knew there was a better way to present this than the pair of comments I dropped.

Reply:
Implemented your suggestion in [969e56b](https://github.com/WebKit/WebKit/commit/969e56be3a56363a7cd3acee00649dbe12ef1814) through:

```cpp
enum class ActiveStateChangeReason : uint8_t {
    HandledContextMenuEvent,
    BeganTrackingSelection,
    SetCurrentSelection,
    WindowActivityChanged,
};

void repaintOnSelectionActiveStateChangeIfNeeded(ActiveStateChangeReason);
```

The selection-related paint request logic is much cleaner now!